File: examples/python/01_pull_bundle.py
"""Example: Pull the demo-starter bundle and inspect its contents."""

import musher

# Credentials are auto-discovered from the MUSHER_API_KEY environment variable,
# the OS keyring, or a credential file (~/.config/musher/credentials.yaml).
# To set credentials explicitly: musher.configure(token="mush_...")

bundle = musher.pull("musher-examples/demo-starter:1.0.0")

print(f"Bundle: {bundle.ref} v{bundle.version}")
print()

# List all files in the bundle
print(f"Files ({len(list(bundle.files()))}):")
for fh in bundle.files():
    print(f"  {fh.logical_path}  ({fh.media_type or 'unknown'})")
print()

# List available skills
print(f"Skills ({len(bundle.skills())}):")
for skill in bundle.skills():
    print(f"  {skill.name}")
    # Show a preview of the skill's SKILL.md content
    preview = skill.skill_md().text().strip().splitlines()
    # Skip frontmatter, show first content line
    in_frontmatter = False
    for line in preview:
        if line.strip() == "---":
            in_frontmatter = not in_frontmatter
            continue
        if not in_frontmatter and line.strip():
            print(f"    → {line.strip()[:100]}")
            break
print()

# List available agents
print(f"Agent specs ({len(bundle.agent_specs())}):")
for agent in bundle.agent_specs():
    print(f"  {agent.name}")
print()

# Access a specific skill by name
try:
    skill = bundle.skill("summarizing-changes")
    print(f"Skill content preview ({skill.name}):")
    print(skill.skill_md().text()[:300])
    print("...")
except KeyError:
    print("Skill 'summarizing-changes' not found in bundle")

File: examples/typescript/01-pull-bundle.ts
/**
 * Pull the demo-starter bundle and inspect its contents.
 *
 * Prerequisites:
 *   export MUSHER_API_KEY="mush_..."
 *
 * Run:
 *   npx tsx 01-pull-bundle.ts
 */

import { pull } from "@musher-dev/musher-sdk";

const bundle = await pull("musher-examples/demo-starter:1.0.0");

console.log(`Bundle: ${bundle.ref.toBaseRef()} v${bundle.version}`);
console.log();

// List all files in the bundle
const files = bundle.files();
console.log(`Files (${files.length}):`);
for (const file of files) {
	console.log(`  ${file.logicalPath}  (${file.assetType}, ${file.sizeBytes} bytes)`);
}
console.log();

// List available skills
const skills = bundle.skills();
console.log(`Skills (${skills.length}):`);
for (const skill of skills) {
	console.log(`  ${skill.name}`);
}
console.log();

// List available agent specs
const agentSpecs = bundle.agentSpecs();
console.log(`Agent Specs (${agentSpecs.length}):`);
for (const agentSpec of agentSpecs) {
	console.log(`  ${agentSpec.name}`);
}
console.log();

// Access a specific skill by name
const skill = bundle.skill("summarizing-changes");
if (skill) {
	const preview = skill.definition()!.text().slice(0, 300);
	console.log(`Skill content preview (${skill.name}):`);
	console.log(preview);
	console.log("...");
}
